src/ast/mod.rs

```diff
@@ -2175,6 +2191,100 @@ impl fmt::Display for Statement {
                 write!(f, "ON {} ", on)?;
                 write!(f, "{}", display_separated(clauses, " "))
             }
+            Statement::CreateSequence {
+                temporary,
+                if_not_exists,
+                name,
+                data_type,
+                sequence_options,
+                owned_by,
+            } => {
+                let mut as_type: String = String::from("");
+                if let Some(dt) = data_type.as_ref() {
+                    as_type = " AS ".to_string() + &dt.to_string();
+                }
```

Comment:
As written this looks like it will do a bunch of string creation and copying that is unnecessary. Perhaps something like this would be less code and more efficient

```suggestion
                let as_type =  if let Some(dt) = data_type.as_ref() {
                    format!(" AS {}", dt);
                } else {
                    "".to_string()
                }
```